aten/src/ATen/native/native_functions.yaml

```diff
@@ -1606,134 +1640,131 @@
 - func: _dimI(Tensor self) -> int64_t
   variants: method
-  dispatch: _sparseDims
+  dispatch: _sparseDims_sparse
+  device_guard: False
 
 
-- func: _native_denseDims(Tensor self) -> int64_t
-  variants: function
-  dispatch:
-    SparseCPU: _denseDims_sparse
-
 - func: _denseDims(Tensor self) -> int64_t
   variants: method
+  dispatch:
+    SparseCPU: _denseDims_sparse
+    SparseCUDA: _denseDims_sparse
+  device_guard: False
 
 # legacy method
 - func: _dimV(Tensor self) -> int64_t
   variants: method
-  dispatch: _denseDims
+  dispatch: _denseDims_sparse
+  device_guard: False
 
 
-- func: _native_nnz(Tensor self) -> int64_t
-  variants: function
-  dispatch:
-    SparseCPU: _nnz_sparse
-
 - func: _nnz(Tensor self) -> int64_t
   variants: method
-
-
-- func: native_coalesce(Tensor self) -> Tensor
-  variants: function
   dispatch:
-    SparseCPU: coalesce_sparse_cpu
+    SparseCPU: _nnz_sparse
+    SparseCUDA: _nnz_sparse
+  device_guard: False
+
 
 - func: coalesce(Tensor self) -> Tensor
   variants: method
-
-
-- func: native_is_coalesced(Tensor self) -> bool
-  variants: function
   dispatch:
-    SparseCPU: is_coalesced_sparse
+    SparseCPU: coalesce_sparse_cpu
+    SparseCUDA: coalesce_sparse_cuda
+
 
 - func: is_coalesced(Tensor self) -> bool
   variants: method
-
-
-- func: _native_indices(Tensor self) -> Tensor
-  variants: function
   dispatch:
-    SparseCPU: _indices_sparse
+    SparseCPU: is_coalesced_sparse
+    SparseCUDA: is_coalesced_sparse
+  device_guard: False
+
 
 - func: _indices(Tensor self) -> Tensor
   variants: method
-
-
-- func: _native_values(Tensor self) -> Tensor
-  variants: function
   dispatch:
-    SparseCPU: _values_sparse
+    SparseCPU: _indices_sparse
+    SparseCUDA: _indices_sparse
+  device_guard: False
+
 
 - func: _values(Tensor self) -> Tensor
   variants: method
+  dispatch:
+    SparseCPU: _values_sparse
+    SparseCUDA: _values_sparse
+  device_guard: False
 
 
-- func: native_hspmm_out(Tensor result, Tensor mat1, Tensor mat2) -> Tensor
+- func: hspmm_out(Tensor result, Tensor mat1, Tensor mat2) -> Tensor
   variants: function
   dispatch:
     SparseCPU: hspmm_out_sparse_cpu
+    SparseCUDA: hspmm_out_sparse_cuda
 
-- func: native_hspmm(Tensor mat1, Tensor mat2) -> Tensor
+- func: hspmm(Tensor mat1, Tensor mat2) -> Tensor
   variants: function
   dispatch:
     SparseCPU: hspmm_sparse_cpu
-
-- func: hspmm_out(Tensor result, Tensor mat1, Tensor mat2) -> Tensor
-  variants: function
-
-- func: hspmm(Tensor mat1, Tensor mat2) -> Tensor
-  variants: function
+    SparseCUDA: hspmm_sparse_cuda
 
 # This "raw copy" doesn't handle conversions NOR does it handle non-blocking.
 - func: raw_copy_sparse_(Tensor self, Tensor src) -> Tensor
   variants: function
   dispatch:
     SparseCPU: copy_sparse_
+    SparseCUDA: copy_sparse_
 
 - func: numel(Tensor self) -> int64_t
   variants:
     - method
     - function
+  device_guard: False
+
+
+- func: native_get_device(Tensor self) -> int64_t
```

Comment:
am I missing something?  this function isn't defined?

Reply:
It's got a dispatch, it doesn't have to be defined:

```
- func: native_get_device(Tensor self) -> int64_t
  variants: function
  dispatch:
    SparseCUDA: get_device_sparse_cuda
  device_guard: False
```